ios/browser/api/p3a/brave_histograms_controller.mm

```diff
@@ -0,0 +1,114 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/ios/browser/api/p3a/brave_histograms_controller.h"
+
+#include "base/strings/sys_string_conversions.h"
+#include "ios/chrome/browser/browser_state/chrome_browser_state.h"
+#include "ios/chrome/browser/chrome_url_constants.h"
+#include "ios/components/webui/web_ui_url_constants.h"
+#include "ios/web/public/browser_state.h"
+#include "ios/web/public/navigation/navigation_manager.h"
+#include "ios/web/public/web_state.h"
+#include "ios/web/public/web_state_delegate_bridge.h"
+#include "ios/web/public/web_state_observer_bridge.h"
+#include "ios/web/public/webui/web_ui_ios.h"
+#include "ios/web/web_state/ui/crw_web_controller.h"
+#include "ios/web/web_state/ui/wk_web_view_configuration_provider.h"
+#include "ios/web/web_state/web_state_impl.h"
+#include "ios/web/webui/web_ui_ios_impl.h"
+#include "net/base/mac/url_conversions.h"
+
+#if !defined(__has_feature) || !__has_feature(objc_arc)
+#error "This file requires ARC support."
+#endif
+
+@interface BraveHistogramsController () <CRWWebStateDelegate,
+                                         CRWWebStateObserver> {
+  std::unique_ptr<web::WebState> _webState;
+  std::unique_ptr<web::WebStateObserverBridge> _webStateObserver;
+  std::unique_ptr<web::WebStateDelegateBridge> _webStateDelegate;
+
+  WKWebView* _webView;
+  CRWWebController* _webController;
```

Comment:
BraveHistogramsController will be used similar way like sync-internals right ?

Reply:
Yep! Exact same